tests/test_address_manager.py

```diff
@@ -0,0 +1,198 @@
+        miner_address: Address = get_miner_address_from_credentials(AddressType.IP, port)
+        if create_state_manager:
+            self.state_manager: MemoryMinerShieldStateManager = MemoryMinerShieldStateManager()
+        return AwsAddressManager(aws_access_key_id, aws_secret_access_key, miner_region_name=miner_region_name,
+                                 miner_address=miner_address, hosted_zone_id=hosted_zone_id,
+                                 event_processor=PrintingMinerShieldEventProcessor(), state_manager=self.state_manager)
+
+    def test_create_elb(self):
+        """ Test creating ELB by AwsAddressManager class. """
+        address_manager: AwsAddressManager = self.create_aws_address_manager()
+
+        try:
+            # This triggers creation of ELB
+            address_manager.validate_addresses(MappingProxyType({}))
+
+            state: MinerShieldState = self.state_manager.get_state()
+            address_manager_state: MappingProxyType[str, str] = state.address_manager_state
+            assert address_manager_state[address_manager.HOSTED_ZONE_ID_KEY] == aws_route53_hosted_zone_id
+            assert address_manager_state[address_manager.INSTANCE_ID_KEY] == miner_instance_id
+            assert int(address_manager_state[address_manager.INSTANCE_PORT_KEY]) == miner_instance_port
+            created_objects: MappingProxyType[str, frozenset[str]] = state.address_manager_created_objects
+            assert len(created_objects[AwsObjectTypes.WAF.value]) == 1
+            assert len(created_objects[AwsObjectTypes.ELB.value]) == 1
+            assert len(created_objects[AwsObjectTypes.SUBNET.value]) == 1
+            assert len(created_objects[AwsObjectTypes.TARGET_GROUP.value]) == 1
+            assert len(created_objects[AwsObjectTypes.SECURITY_GROUP.value]) == 1
+            assert AwsObjectTypes.DNS_ENTRY.value not in created_objects
+
+            reloaded_state: MinerShieldState = self.state_manager.get_state(reload=True)
+            assert reloaded_state == state
+
+            # Call clean_all and check if everything was removed
+            address_manager.clean_all()
+            state = self.state_manager.get_state()
+            created_objects = state.address_manager_created_objects
+            assert AwsObjectTypes.WAF.value not in created_objects
+            assert AwsObjectTypes.ELB.value not in created_objects
+            assert AwsObjectTypes.SUBNET.value not in created_objects
+            assert AwsObjectTypes.TARGET_GROUP.value not in created_objects
+            assert AwsObjectTypes.SECURITY_GROUP.value not in created_objects
+            assert AwsObjectTypes.DNS_ENTRY.value not in created_objects
+        finally:
+            address_manager.clean_all()
+
+    def test_handle_address(self):
+        """ Create address, validate it and remove created address. """
+        address_manager: AwsAddressManager = self.create_aws_address_manager()
+
+        try:
```

Comment:
try to use https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#yield-fixtures-recommended for the address_manager creation & cleanuup

Reply:
Ok